live-build/misc/ansible-roles/appliance-build.dcenter/tasks/main.yml

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+- include: ldap.yml
+
+- apt:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - python-pip
+    - libldap2-dev
+    - libsasl2-dev
+    - isc-dhcp-server
+    - bind9
+    - dnsutils
+    - telnet
+  register: result
+  until: result is not failed
+  retries: 3
+  delay: 60
+
+
+- pip:
+    name: pysphere
+    extra_args: --trusted-host devpi.delphix.com -i http://devpi.delphix.com:3141/root/public/+simple/
+    version: 0.1.8
+
+- pip:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - paramiko
+    - python-ldap
+    - requests
+  register: result
+  until: result is not failed
+  retries: 3
+  delay: 60
+
+- git:
+    repo: "{{ item.repo }}"
+    dest:
+      "/opt/dcenter/lib/{{ item.dest }}"
+    version: "{{ item.version }}"
+    accept_hostkey: yes
+    update: no
+  with_items:
+    - { repo: 'https://gitlab.delphix.com/devops/dcenter-gate.git',
+        version: projects/dx4linux,
+        dest: dcenter-gate }
+
+- copy:
+    dest: "/etc/dcenter.conf"
+    content: |
+      [[ -z "$DC_ROOT" ]] && export DC_ROOT="dcenter"
+
+- copy:
+    dest: "/etc/apparmor.d/local/usr.sbin.dhcpd"
```

Comment:
Why do we need to play with apparmor? (Could you also add some comments)

Reply:
fixed